src/jit/compiler.cpp

```diff
@@ -4999,8 +4999,12 @@ int           Compiler::compCompileHelper (CORINFO_MODULE_HANDLE            clas
             (fgBBcount > 5) &&
             !forceInline)
         {
-            compInlineResult->setNever("Too many basic blocks in the inlinee");
-            goto _Next;
+            compInlineResult->note(InlineObservation::CALLEE_TOO_MANY_BASIC_BLOCKS);
+
+            if (compInlineResult->isFailure()) 
+            {
+                goto _Next;
+            }
```

Comment:
These two (or maybe it's three?) patterns of usage are a bit confusing to me. It would appear that in some cases the code is "asserting fatality", in others it is making an observation and the querying about whether this (or some other?) observation resulted in failure, and then perhaps in others (?) it is merely making an observation. It would seem better to always use the "note, then query" form. Am I missing something?